pkg/skaffold/build/buildpacks/dependencies.go

```diff
@@ -33,8 +34,27 @@ func GetDependencies(ctx context.Context, workspace string, a *latest.BuildpackA
 		return nil, errors.Wrapf(err, "walking workspace %s", workspace)
 	}
 
-	var dependencies []string
+	// Make sure folders are also listed
+	added := map[string]bool{}
 	for file := range files {
+		for path := file; path != "."; path = filepath.Dir(path) {
+			added[path] = true
+		}
+	}
+
+	for path := range files {
+		for path != "." {
+			if added[path] {
+				break
+			}
+
+			added[path] = true
+			path = filepath.Dir(path)
+		}
+	}
```

Comment:
Isn't this effectively the same as the code above?

Reply:
lol. there are the old and the new optimized version...